### PR TITLE
Detect orphaned machines

### PR DESCRIPTION
**Pull request by mattmattox**
This script is designed to walk through all clusters in a Rancher install to find orphaned machine objects.

NOTE: This script is read-only does not make any changes to the cluster, node, or Rancher.

Script logic:
This script connects to the Rancher API using a key and the Rancher URL. It will then create a list of all clusters. It will then generate a kubeconfig for each cluster. It then walks through each machine object and tries to match it to a node in the cluster using kubectl get nodes.

### CODE REVIEW

File: out-of-sync-global-dns/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+### Setup credentials and deploy the workload
+- Please run the following to create a namespace
+```
+kubectl create namespace out-of-sync-global-dns
+```
+
+- Please run the following to create a secret using the API keys created earlier.
+```
+kubectl -n out-of-sync-global-dns create secret generic rancher-api-key \
+  --from-literal=cattle-server="https://rancher.example.com/v3" \
```
Comment (ansilh) [inline out-of-sync-global-dns/README.md:30]: Only FQDN is need in `cattle-server`

File: out-of-sync-global-dns/workload.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (ansilh) [inline out-of-sync-global-dns/workload.yaml:1]: Add a service account and RBAC to allow the pod to access the needed resources

File: out-of-sync-global-dns/workload.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+spec:
+  selector:
+    matchLabels:
+      app: out-of-sync-global-dns
+  template:
+    metadata:
+      labels:
+        app: out-of-sync-global-dns
+    spec:
+      containers:
```
Comment (ansilh) [inline out-of-sync-global-dns/workload.yaml:17]: Need to add `args` to execute the script

**Review by ansilh — CHANGES_REQUESTED** [review]
Reviewed `detect-orphaned-machines` 
- Added rbac to the workload.yaml 
- Need to change `cattle-server` flag to include only FQDN 

Please add RBAC to ` out-of-sync-global-dns` and modify workload to include the script.

`Deployment` will cause the Pod to restart after script execution causing multiple execution (loop)

Can we change the workload to `Job` instead of `Deployment` if this needs to be executed only once?

File: out-of-sync-global-dns/README.md (hunk 1 above)
Comment (mattmattox) [reply out-of-sync-global-dns/README.md:30]: Fixed

File: out-of-sync-global-dns/workload.yaml (hunk 2 above)
Comment (mattmattox) [reply out-of-sync-global-dns/workload.yaml:1]: Fixed.
serviceaccount/detect-orphaned-machines created
clusterrole.rbac.authorization.k8s.io/detect-orphaned-machines created
clusterrolebinding.rbac.authorization.k8s.io/detect-orphaned-machines created
deployment.apps/detect-orphaned-machines configured

File: out-of-sync-global-dns/workload.yaml (hunk 3 above)
Comment (mattmattox) [reply out-of-sync-global-dns/workload.yaml:17]: Fixed

**Pull request closed without merging**